daemon/daemon.go

```diff
@@ -169,19 +170,20 @@ func (d *Daemon) Run() error {
 	}
 	d.containerMgr = containerMgr
 
+	if err := containerMgr.Restore(ctx); err != nil {
+		return err
+	}
+
+	// ensure all containers have been restored
+	time.Sleep(100 * time.Millisecond)
```

Comment:
`time.Sleep` is a tricky method to wait network init to be finished, how about use a channel to wait ?

Reply:
Indeed, the Sleep is to prevent the situation described #1421.  I want to catch the dead container exit status from statusCh. But the healthy container will never return exit status unless it exited.